packages/portfolio-contract/test/network/prod-network.test.ts

```diff
@@ -54,70 +76,80 @@ const getGraph = () =>
   makeGraphFromDefinition(PROD_NETWORK, {}, {}, brand, feeBrand);
 
 const hasEdge = (
-  edges: ReturnType<typeof getGraph>['edges'],
-  src: string,
-  dest: string,
-  via?: string,
+  edges: RebalanceGraph['edges'],
+  src: AssetPlaceRef,
+  dest: AssetPlaceRef,
+  via?: TransferProtocol,
 ) =>
   edges.some(
-    e =>
-      e.src === (src as any) &&
-      e.dest === (dest as any) &&
-      (via ? e.via === (via as any) : true),
+    e => e.src === src && e.dest === dest && (via ? e.via === via : true),
   );
 
 const isReachable = (
-  edges: ReturnType<typeof getGraph>['edges'],
-  start: string,
-  goal: string,
+  edges: RebalanceGraph['edges'],
+  start: AssetPlaceRef,
+  goal: AssetPlaceRef,
 ) => {
-  const q = [start as any];
-  const seen = new Set<string>([start]);
+  const q = [start];
+  const seen = new Set<AssetPlaceRef>([start]);
   while (q.length) {
-    const cur = q.shift() as any;
+    const cur = q.shift()!;
     if (cur === goal) return true;
     for (const e of edges) {
       if (e.src !== cur) continue;
-      if (!seen.has(e.dest as any)) {
-        seen.add(e.dest as any);
-        q.push(e.dest as any);
+      if (!seen.has(e.dest)) {
+        seen.add(e.dest);
+        q.push(e.dest);
       }
     }
   }
   return false;
 };
 
 // Map a pool key to its hub based on naming convention (or noble for USDN)
-const poolToHub = (pool: string): AssetPlaceRef => {
-  if (pool === 'USDN' || pool === 'USDNVault') return '@noble';
-  const parts = pool.split('_');
-  const chain = parts[parts.length - 1];
-  return `@${chain}` as AssetPlaceRef;
-};
+const poolToHub = (pool: PoolKey): AssetPlaceRef =>
+  `@${PoolPlaces[pool].chainName}`;
 
 // Derived intra-chain validation pairs
-const INTRA_PAIRS: Array<[string, AssetPlaceRef]> = POOLS.map(p => [
+const INTRA_PAIRS: Array<[PoolKey, AssetPlaceRef]> = POOLS.map(p => [
   p,
   poolToHub(p),
 ]);
 
 // Tests
 
+// Parity checks with source definitions
+test('HUBS matches SupportedChain values', t => {
+  t.deepEqual(
+    Object.keys(SupportedChain).sort(),
+    HUBS.map(h => h.slice(1)).sort(),
+  );
```

Comment:
`t.deepEqual` is really bad at reporting diffs on arrays. It does much better if you turn them into `{ item1: null, item2: null, ...}` objects.